input/script_utils.go

```diff
@@ -1504,6 +1504,61 @@ func TaprootSecondLevelHtlcScript(revokeKey, delayKey *btcec.PublicKey,
 	return redemptionKey, nil
 }
 
+// SecondLevelScriptTree is a tapscript tree used to spend the second level
+// HTLC output after the CSV delay has passed.
+type SecondLevelScriptTree struct {
```

Comment:
could probs have like a `BaseScriptTree` which can then be embedded in here and in `CommitScriptTree`